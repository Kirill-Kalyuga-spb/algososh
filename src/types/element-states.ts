export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
}

export type TString = {
  letter: string;
  state: ElementStates;
}

export type TSorting = {
  index: number;
  state: ElementStates;
}