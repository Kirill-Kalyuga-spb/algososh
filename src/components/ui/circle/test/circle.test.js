import renderer from 'react-test-renderer'
import { Circle } from '../circle'
import { ElementStates } from '../../../../types/element-states';

describe('Тест компонента Circle', () => {
    it('без буквы', () => {
        const tree = renderer
        .create(<Circle letter='' />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('с буквами', () => {
        const tree = renderer
        .create(<Circle letter='ABC' />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('с head', () => {
        const tree = renderer
        .create(<Circle head='head' />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('с react-элементом в head', () => {
        const tree = renderer
        .create(<Circle head={<Circle />} />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('с tail', () => {
        const tree = renderer
        .create(<Circle tail='tail' />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('с react-элементом в tail', () => {
        const tree = renderer
        .create(<Circle tail={<Circle />} />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('с index', () => {
        const tree = renderer
        .create(<Circle index={1} />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('с пропом isSmall ===  true', () => {
        const tree = renderer
        .create(<Circle isSmall={true} />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('в состоянии default', () => {
        const tree = renderer
        .create(<Circle state={ElementStates.Default} />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('в состоянии changing', () => {
        const tree = renderer
        .create(<Circle state={ElementStates.Changing} />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('в состоянии modified', () => {
        const tree = renderer
        .create(<Circle state={ElementStates.Modified} />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
})