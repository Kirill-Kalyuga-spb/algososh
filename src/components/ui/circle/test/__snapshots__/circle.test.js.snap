// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента Circle без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        element"
    data-testid="head"
  />
  <div
    className="circle  
         
        default"
    data-testid="circle-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4
        absolute 
        tail30 
        element"
    data-testid="tail"
  />
</div>
`;

exports[`Тест компонента Circle в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        element"
    data-testid="head"
  />
  <div
    className="circle  
         
        changing"
    data-testid="circle-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4
        absolute 
        tail30 
        element"
    data-testid="tail"
  />
</div>
`;

exports[`Тест компонента Circle в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        element"
    data-testid="head"
  />
  <div
    className="circle  
         
        default"
    data-testid="circle-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4
        absolute 
        tail30 
        element"
    data-testid="tail"
  />
</div>
`;

exports[`Тест компонента Circle в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        element"
    data-testid="head"
  />
  <div
    className="circle  
         
        modified"
    data-testid="circle-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4
        absolute 
        tail30 
        element"
    data-testid="tail"
  />
</div>
`;

exports[`Тест компонента Circle с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        string"
    data-testid="head"
  >
    head
  </div>
  <div
    className="circle  
         
        default"
    data-testid="circle-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4
        absolute 
        tail30 
        element"
    data-testid="tail"
  />
</div>
`;

exports[`Тест компонента Circle с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        element"
    data-testid="head"
  />
  <div
    className="circle  
         
        default"
    data-testid="circle-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4
        absolute 
        tail60 
        element"
    data-testid="tail"
  />
</div>
`;

exports[`Тест компонента Circle с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        element"
    data-testid="head"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        element"
        data-testid="head"
      />
      <div
        className="circle  
         
        default"
        data-testid="circle-circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4
        absolute 
        tail30 
        element"
        data-testid="tail"
      />
    </div>
  </div>
  <div
    className="circle  
         
        default"
    data-testid="circle-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4
        absolute 
        tail30 
        element"
    data-testid="tail"
  />
</div>
`;

exports[`Тест компонента Circle с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        element"
    data-testid="head"
  />
  <div
    className="circle  
         
        default"
    data-testid="circle-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4
        absolute 
        tail30 
        element"
    data-testid="tail"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        element"
        data-testid="head"
      />
      <div
        className="circle  
         
        default"
        data-testid="circle-circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4
        absolute 
        tail30 
        element"
        data-testid="tail"
      />
    </div>
  </div>
</div>
`;

exports[`Тест компонента Circle с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        element"
    data-testid="head"
  />
  <div
    className="circle  
         
        default"
    data-testid="circle-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4
        absolute 
        tail30 
        string"
    data-testid="tail"
  >
    tail
  </div>
</div>
`;

exports[`Тест компонента Circle с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        element"
    data-testid="head"
  />
  <div
    className="circle  
         
        default"
    data-testid="circle-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4
        absolute 
        tail30 
        element"
    data-testid="tail"
  />
</div>
`;

exports[`Тест компонента Circle с пропом isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
        absolute 
        head 
        element"
    data-testid="head"
  />
  <div
    className="circle  
        small 
        default"
    data-testid="circle-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4
        absolute 
        tail30 
        element"
    data-testid="tail"
  />
</div>
`;
